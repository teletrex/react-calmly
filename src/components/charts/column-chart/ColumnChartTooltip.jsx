/* <LICENSE>
*
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
*
* </LICENSE>
*  */



import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import  settings  from '../../../settings';

const { prefix } = settings;

const ColumnChartTooltip = ({ data, label }) => {
  const barData = useMemo(() => {
    if (label) {
      const singleBarData = data.filter(item => item.name === label);
      return singleBarData[0];
    }
    return null;
  }, [data, label]);

  if (barData) {
    const { name, value } = barData;

    return (
      <div className={`${prefix}--column-chart__tooltip`}>
        <div className={`${prefix}--column-chart__tooltip-name`}>{name}</div>
        <div className={`${prefix}--column-chart__tooltip-value`}>{value}</div>
      </div>
    );
  }

  return null;
};

ColumnChartTooltip.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired,
      value: PropTypes.number.isRequired,
    })
  ),
  label: PropTypes.string,
};

ColumnChartTooltip.defaultProps = {
  data: [],
  label: '',
};

export default ColumnChartTooltip;
