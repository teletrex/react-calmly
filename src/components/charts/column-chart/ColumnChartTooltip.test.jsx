/* <LICENSE>
* 
* Copyright (C) 2022 Louis F. Roehrs, All rights reserved.
* 
* </LICENSE>
*  */



import React from 'react';
import { render } from '@testing-library/react';

import ColumnChartTooltip from './ColumnChartTooltip';

const data = [
  { name: '2016 Q1', value: 98, color: '#29ab87' },
  { name: '2016 Q2', value: 96, color: '#0292d4' },
  { name: '2016 Q3', value: 94, color: '#d846bf' },
  { name: '2016 Q4', value: 93, color: '#f45d20' },
  { name: '2017 Q1', value: 90, color: '#f3ba00' },
  { name: '2017 Q2', value: 88, color: '#ff465a' },
  { name: '2017 Q3', value: 87, color: '#FF8C7B' },
  { name: '2017 Q4', value: 86, color: '#29ab87' },
  { name: '2018 Q1', value: 84, color: '#0292d4' },
  { name: '2018 Q2', value: 80, color: '#d846bf' },
  { name: '2018 Q3', value: 77, color: '#f45d20' },
  { name: '2018 Q4', value: 75, color: '#f3ba00' },
  { name: '2019 Q1', value: 74, color: '#ff465a' },
  { name: '2019 Q2', value: 72, color: '#FF8C7B' },
];

describe('ColumnChartTooltip', () => {
  test('verify if created', () => {
    const { container } = render(<ColumnChartTooltip data={data} label="2017 Q4" />);
    expect(container.firstChild).toMatchSnapshot();
  });

  test('returns null if label does not match on data', () => {
    const { container } = render(<ColumnChartTooltip data={data} />);
    expect(container.childElementCount).toBe(0);
  });

  test('Uses correct data name and value', () => {
    const { getByText } = render(<ColumnChartTooltip data={data} label="2017 Q3" />);
    expect(getByText('87')).not.toBeFalsy();
    expect(getByText('2017 Q3')).not.toBeFalsy();
  });
});
