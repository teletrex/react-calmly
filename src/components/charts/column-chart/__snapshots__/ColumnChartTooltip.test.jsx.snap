// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColumnChartTooltip verify if created 1`] = `
<div
  class="bx--column-chart__tooltip"
>
  <div
    class="bx--column-chart__tooltip-name"
  >
    2017 Q4
  </div>
  <div
    class="bx--column-chart__tooltip-value"
  >
    86
  </div>
</div>
`;
